gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/InsertAllResponse.java

```diff
@@ -0,0 +1,119 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.bigquery;
+
+import com.google.api.services.bigquery.model.ErrorProto;
+import com.google.api.services.bigquery.model.TableDataInsertAllResponse;
+import com.google.api.services.bigquery.model.TableDataInsertAllResponse.InsertErrors;
+import com.google.common.base.Function;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Google Cloud BigQuery insert all response. Objects of this class possibly contain errors for an
+ * {@link InsertAllRequest}. If a row failed to be inserted, the non-empty list of errors associated
+ * to that row's index can be obtained with {@link InsertAllResponse#errorsFor(Long)}.
+ * {@link InsertAllResponse#insertErrors()} can be used to return all errors caused by a
+ * {@link InsertAllRequest} as a map.
+ */
+public class InsertAllResponse implements Serializable {
+
+  private final Map<Long, List<BigQueryError>> insertErrors;
+
+  InsertAllResponse(Map<Long, List<BigQueryError>> insertErrors) {
+    this.insertErrors = insertErrors != null ? ImmutableMap.copyOf(insertErrors)
+        : ImmutableMap.<Long, List<BigQueryError>>of();
+  }
+
+  /**
+   * Returns all insertion errors as a map whose keys are indexes of rows that failed to insert.
+   * Each failed row index is associated with a non-empty list of {@link BigQueryError}.
+   */
+  public Map<Long, List<BigQueryError>> insertErrors() {
+    return insertErrors;
+  }
+
+  /**
+   * Returns errors for the provided row index. If no error exists returns {@code null}.
+   */
+  public List<BigQueryError> errorsFor(Long index) {
+    return insertErrors.get(index);
+  }
+
+  /**
+   * Returns {@code true} if no row insertion failed, {@code false} otherwise. If {@code false}
+   * {@link #insertErrors()} returns an empty map.
+   */
+  public boolean hasErrors() {
+    return !insertErrors.isEmpty();
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(insertErrors);
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    return obj instanceof InsertAllResponse
+        && Objects.equals(insertErrors, ((InsertAllResponse) obj).insertErrors);
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this).add("insertErrors", insertErrors).toString();
+  }
+
+  TableDataInsertAllResponse toPb() {
```

Comment:
Yes, the conversions from and to `TableDataInsertAllResponse` are more complex than usual so I added this for easier testing. I could remove it though.